plugins/permission/src/components/WithPermission.test.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Permission } from '@backstage/plugin-permission-common';
+import { render } from '@testing-library/react';
+import { WithPermission } from './WithPermission';
+import { usePermission } from '../hooks/usePermission';
+
+jest.mock('../hooks/usePermission', () => ({
+  AsyncPermissionResult: jest.requireActual('../hooks/usePermission')
+    .AsyncPermissionResult,
+  usePermission: jest.fn(),
+}));
```

Comment:
nit: since `AsyncPermissionResult` is just a type now, I don't think we need to include it in our mock.

```suggestion
jest.mock('../hooks/usePermission', () => ({
  usePermission: jest.fn(),
}));
```